sympy/logic/boolalg.py

```diff
@@ -1004,6 +1010,16 @@ def to_nnf(self, simplify=True):
                 args.append(Or(*clause))
         return And._to_nnf(*args, simplify=simplify)
 
+    def _eval_rewrite_as_Or(self, *args, **kwargs):
+        a = self.args
+        return Or(*[_convert_to_varsSOP(x, self.args)
+                    for x in _get_odd_parity_terms(len(a))])
+
+    def _eval_rewrite_as_And(self, *args, **kwargs):
+        a = self.args
+        return And(*[_convert_to_varsPOS(x, self.args)
+                     for x in _get_even_parity_terms(len(a))])
```

Comment:
Are these two functions converting to CNF/DNF (SOP/POS)? If so is that not already implemented?

Reply:
See this comment: https://github.com/sympy/sympy/pull/16194#discussion_r296863743

Basically, it has to do with the expected form. Rewrite rewrites using a closed form expression, keeping sub-expressions, while to_cnf/to_nnf tried to simplify the expression (to some extent) and keeping everything on an SOP/POS form (but not simplifying it to the minimal SOP/POS form). I think both have their use.

Reply:
Sounds like you're more qualified than me to merge this :)

Reply:
My only comment would be that it would be good to add tests that demonstrate the similarity/difference between this and the related methods.

Reply:
I can do that in another PR where I can more easily run the tests locally.

Reply:
I have the following in my `.git/config`:
```
[remote "upstream"]
    url = https://github.com/sympy/sympy.git
    fetch = +refs/heads/*:refs/remotes/upstream/*
    fetch = +refs/pull/*/head:refs/remotes/upstream/pr/*
```
The means that `git fetch upstream` gives me a local copy of every PR. I can then check this out with
```console
$ git fetch upstream
$ git checkout upstream/pr/16194
```
The first time you run git fetch with this is slow.

I'm still not quite sure what is the best way to push to a PR using this setup...

Reply:
Thanks for the hint! Indeed a good way to try out PRs I believe.

I'll merge this for now then and submit a new PR with more tests.